src/main/java/com/ibm/devops/connect/CRPipeline/UploadASoCTestResult.java

```diff
@@ -109,6 +113,30 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         String buildUrlValue = envVars.expand(this.buildUrl);
         String metricDefinitionValue = envVars.expand(this.metricDefinition);
         String recordNameValue = envVars.expand(this.recordName);
+        String instanceBaseUrl = envVars.expand(this.instanceBaseUrl == null ? "" : this.instanceBaseUrl.toString());
+
+        List<Entry> entries = Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getEntries();
+        int instanceNum = 0;
+        if(instanceBaseUrl.equals("Upload ASoC Test Result to All UCV Instances")){
+            instanceNum = -1;
+        }else if(StringUtils.isNotEmpty(instanceBaseUrl)){
+            try {
+                int i=0;
+                for (Entry entry : entries) {
+                    if(removeTrailingSlash(instanceBaseUrl).equals(removeTrailingSlash(entry.getBaseUrl()))){
```

Comment:
Was there any consideration to allowing them to name the instance so we could match with that?  I guess that could be an issue if there is an upgrade to this version where an existing consideration does not have a name. Maybe there is a way to set some default name for existing instances?  I feel like trying to match name is easier than trying to match url.